core/src/main/scala/org/apache/spark/ui/JettyUtils.scala

```diff
@@ -270,9 +270,21 @@ private[spark] object JettyUtils extends Logging {
 
       gzipHandlers.foreach(collection.addHandler)
       connectors.foreach(_.setHost(hostName))
+      // As each Acceptor will use one thread, the number of threads should at least be the number
+      // of acceptors plus 1.
+      val minThreads =
+        connectors.collect { case c: AbstractConnector => c }.map(_.getAcceptors).sum + 1
       server.setConnectors(connectors.toArray)
 
       val pool = new QueuedThreadPool
+      var maxThreads = conf.getInt("spark.ui.threads", pool.getMaxThreads)
```

Comment:
Agree, if it's just for web UIs then I see no need for a large number of acceptors. I think your change is still needed to make sure the pool has enough threads in any event? though maybe the config is unnecessary.